### ISSUE TITLE
Build on Windows using MSYS2?

### ISSUE THREAD

**Issue by lxkurko** [openlierox/openlierox#826]
I think it would be useful if Openlierox could be built on Windows using MSYS2+MinGW-w64. That way one could use the same tools as on Linux/Unix systems, and dependency management would be easier.

**Comment by albertz** [comment]
Yes, definitely. I think actually we tried and tested that, and it worked (I don't remember how exactly; maybe it just worked right out of the box via CMake). However, as far as I remember, Karel said that the Visual Studio build had better performance (I mean the resulting openlierox.exe), and that is what matters most in the end.

**Comment by albertz** [comment]
Ah, and another thing, which also matters: MSys2/MingW (I think using GCC) created different debug symbols than the normal VS build would do. And the crash handler which we used, which creates a nice human readable back trace, only worked with the VC debug symbols.

**Comment by albertz** [comment]
@karelp maybe you also remember some other details.

**Comment by lxkurko** [comment]
Are there any configuration options that should be set when compiling (trying to compile) with MSYS2? I tried it quickly but it seems that it either fails to find the required headers or to identify the system properly.

**Comment by karelp** [comment]
I've never tried to compile with MSYS2. As far as I remember, there were some _MSC_VER checks for Visual Studio in the code - some of them could be relevant to Windows as well. The header search paths will need to be transferred from the VS solution and you will also need proper static libraries (compiled for MingW) for all the dependencies. All in all I expect it to be a lot of work and the toolchain is still quite different from standard Unix/Linux tools, not sure it's worth the time investment.

**Comment by StrikerTheHedgefox** [comment]
If you guys could get this going, I'd be very grateful. I'd like to get a build of this compiled. I tried MSVC, but it's broken (According to MSVC, most of the source files are missing or pointing to the wrong path in the solution). 

Also, seems the wiki for build instructions is dead.

**Comment by albertz** [comment]
Supported now.
There is a full MSYS2 MinGW64 build and packaging pipeline
in .github/workflows/package-windows.yml ("Build Windows .exe (MSYS2 MinGW64)"),
using msys2/setup-msys2 with the mingw-w64 toolchain and SDL2 packages.
Closing as implemented.
